Controller/RegistrationFOSUser1Controller.php

```diff
@@ -59,16 +58,25 @@ public function registerAction()
                 $route = $this->get('session')->get('sonata_basket_delivery_redirect');
 
                 if (null !== $route) {
+                    // NEXT_MAJOR: remove the if block
+                    $deprecationMessage = <<<'EOT'
+Setting a redirect url in the sonata_basket_delivery_redirect session variable
+is deprecated since 3.x and will no longer result in a redirection to this url in 4.0.
+EOT;
+                    @trigger_error($deprecationMessage, E_USER_DEPRECATED);
                     $this->get('session')->remove('sonata_basket_delivery_redirect');
                     $url = $this->generateUrl($route);
                 } else {
-                    $url = $this->get('session')->get('sonata_user_redirect_url', $this->generateUrl('sonata_user_profile_show'));
+                    $url = $this->get('session')->get(
+                        'sonata_user_redirect_url',
+                        $this->generateUrl('sonata_user_profile_show')
+                    );
                 }
             }
 
             $this->setFlash('fos_user_success', 'registration.flash.user_created');
 
-            $response = new RedirectResponse($url);
+            $response = $this->redirect($url);
```

Comment:
It's here since [almost forever](https://github.com/symfony/symfony/commit/9c07e46d91b30236d89a98f738b7b7f3e2e5dfd1)